cpp/dolfin/io/XDMFFile.h

```diff
@@ -105,10 +105,11 @@ class XDMFFile : public common::Variable
 #endif
 
   /// Constructor
-  XDMFFile(const std::string filename) : XDMFFile(MPI_COMM_WORLD, filename) {}
+  XDMFFile(const std::string filename)
```

Comment:
Let's remove this constructor from C++. Nice to have  just one way. We can always add a convenience version to Python